python/foglamp/services/core/api/notification.py

```diff
@@ -0,0 +1,515 @@
+             curl -X POST http://localhost:8081/foglamp/notification -d '{"name": "Test Notification", "description":"Test Notification", "rule": "testrule", "channel": "email", "notification_type": "one shot", "enabled": false, "ruleConfig": {}, "deliveryConfig": {}}'
+    """
+    try:
+        data = await request.json()
+        if not isinstance(data, dict):
+            raise ValueError('Data payload must be a valid JSON')
+
+        name = data.get('name', None)
+        description = data.get('description', None)
+        rule = data.get('rule', None)
+        channel = data.get('channel', None)
+        notification_type = data.get('notification_type', None)
+        enabled = data.get('enabled', None)
+        rule_config = data.get('ruleConfig', None)
+        delivery_config = data.get('deliveryConfig', None)
+
+        if name is None:
+            raise web.HTTPBadRequest(reason='Missing name property in payload.')
+        if description is None:
+            raise web.HTTPBadRequest(reason='Missing description property in payload.')
+        if rule is None:
+            raise web.HTTPBadRequest(reason='Missing rule property in payload.')
+        if channel is None:
+            raise web.HTTPBadRequest(reason='Missing channel property in payload.')
+        if notification_type is None:
+            raise web.HTTPBadRequest(reason='Missing notification_type property in payload.')
+
+        if utils.check_reserved(name) is False:
+            raise web.HTTPBadRequest(reason='Invalid name property in payload.')
+        if utils.check_reserved(rule) is False:
+            raise web.HTTPBadRequest(reason='Invalid rule property in payload.')
+        if utils.check_reserved(channel) is False:
+            raise web.HTTPBadRequest(reason='Invalid channel property in payload.')
+        if notification_type not in ["one shot", "retriggered", "toggled"]:
+            raise web.HTTPBadRequest(reason='Invalid notification_type property in payload.')
+
+        if enabled is not None:
+            if enabled not in ['true', 'false', True, False]:
+                raise web.HTTPBadRequest(reason='Only "true", "false", true, false are allowed for value of enabled.')
+        is_enabled = True if ((type(enabled) is str and enabled.lower() in ['true']) or (
+            (type(enabled) is bool and enabled is True))) else False
+
+        storage = connect.get_storage_async()
+        config_mgr = ConfigurationManager(storage)
+        process_name = 'notification_c'
+        script = '["services/notification_c"]'
+
+        # First check if valid plugins have been supplied
+        rule_plugin_config, delivery_plugin_config = await check_plugins(rule, channel)
+
+        # then check that the schedule name is not already registered
+        count = await check_schedules(storage, name)
+        if count != 0:
+            raise web.HTTPBadRequest(reason='A notification with this name already exists.')
+
+        # Create scheduled process
```

Comment:
There is no scheduled process for each notification. Notifications are handled by the notification service, but one notification service can handle multiple notifications.